openlibrary/plugins/openlibrary/api.py

```diff
@@ -3,12 +3,104 @@
 import simplejson
 
 from infogami.utils import delegate
-from openlibrary.core import helpers as h
+from infogami.utils.view import render_template
+from openlibrary.core import lending, helpers as h
+from openlibrary.data import popular
+
+popular_editions = popular.popular
+
 
 def setup():
     # placeholder
     pass
 
+class popular_books(delegate.page):
+    path = '/popular'
+
+    def GET(self):
+        """Returns `limit` popular book tuples of form [ocaid, olid] starting
+        at `start`
+
+        Should be cached through memcached.
+        """
+        i = web.input(start='0', limit='100')
+        start, limit = int(i.start), int(i.limit),
+        books = popular_editions[start:start+limit]
+        result = {
+            'books': books,
+            'limit': limit,
+            'start': start,
+            'next': start + limit
+        } if books else {'error': 'out_of_range'}
+        return delegate.RawText(simplejson.dumps(result),
+                                content_type="application/json")
+
+def format_edition(edition):
+    """This should be moved to a books or carousel model"""
+    from openlibrary.core import ia
+    collections = ia.get_meta_xml(edition.get('ocaid')).get("collection", [])
```

Comment:
Yes, this pre-existing API is completely confusing. It is the archive.org json metadata api which used to be xml.